pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -259,10 +277,30 @@ func (r *ReconcileBareMetalHost) Reconcile(request reconcile.Request) (result re
 	// over when there is an unrecoverable error (tracked through the
 	// error state of the host).
 	if actResult.Dirty() {
-		info.log.Info("saving host status",
+
+		//Repopulate annotation again
+		delete(host.Annotations, "kubectl.kubernetes.io/last-applied-configuration")
+		mediaType := "application/json"
+		infoNew, ok := runtime.SerializerInfoForMediaType(codecFactory.SupportedMediaTypes(), mediaType)
+		if !ok {
+			info.log.Info("Failed to Serialize")
+		} else {
+			encoder := codecFactory.EncoderForVersion(infoNew.Serializer, metal3v1alpha1.SchemeGroupVersion)
+			newAnnotation, _ := runtime.Encode(encoder, runtime.Object(host))
+			if host.Annotations == nil {
+				host.Annotations = make(map[string]string)
+			}
+			host.Annotations["kubectl.kubernetes.io/last-applied-configuration"] = string(newAnnotation)
```

Comment:
That is not the correct annotation to use. you will conflict with Kubectl. Add a new one that is metal3 specific to store the status.

Reply:
Yes. I have a plan to do that. I am using it as a trial basis